src/Std/Time/Internal/UnitVal.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Init.Data
+import Std.Internal.Rat
+
+namespace Std
+namespace Time
+namespace Internal
+open Std.Internal
+open Lean
+
+set_option linter.all true
+
+/--
+A structure representing a unit of a given ratio type `α`.
+-/
+structure UnitVal (α : Rat) where
+
+  /--
+  Value inside the UnitVal Value.
+  -/
+  val : Int
+  deriving Inhabited, BEq
+
+instance : LE (UnitVal x) where
+  le x y := x.val ≤ y.val
+
+instance { x y : UnitVal z }: Decidable (x ≤ y) :=
+  inferInstanceAs (Decidable (x.val ≤ y.val))
+
+namespace UnitVal
+
+/--
+Creates a `UnitVal` from an `Int`.
+-/
+@[inline]
+def ofInt (value : Int) : UnitVal α :=
+  ⟨value⟩
+
+/--
+Creates a `UnitVal` from a `Nat`.
+-/
+@[inline]
+def ofNat (value : Nat) : UnitVal α :=
+  ⟨value⟩
+
+/--
+Converts a `UnitVal` to an `Int`.
+-/
+@[inline]
+def toInt (unit : UnitVal α) : Int :=
+  unit.val
+
+/--
+Multiplies the `UnitVal` by an `Int`, resulting in a new `UnitVal` with an adjusted ratio.
+-/
+@[inline]
+def mul (unit : UnitVal a) (factor : Int) : UnitVal (a / factor) :=
+  ⟨unit.val * factor⟩
+
+/--
+Divides the `UnitVal` by an `Int`, resulting in a new `UnitVal` with an adjusted ratio.
+-/
+@[inline]
+def ediv (unit : UnitVal a) (divisor : Int) : UnitVal (a * divisor) :=
+  ⟨unit.val.ediv divisor⟩
+
+/--
+Divides the `UnitVal` by an `Int`, resulting in a new `UnitVal` with an adjusted ratio.
+-/
+@[inline]
+def div (unit : UnitVal a) (divisor : Int) : UnitVal (a * divisor) :=
+  ⟨unit.val.tdiv divisor⟩
+
+/--
+Adds two `UnitVal` values of the same ratio.
+-/
+@[inline]
+def add (u1 : UnitVal α) (u2 : UnitVal α) : UnitVal α :=
+  ⟨u1.val + u2.val⟩
+
+/--
+Subtracts one `UnitVal` value from another of the same ratio.
+-/
+@[inline]
+def sub (u1 : UnitVal α) (u2 : UnitVal α) : UnitVal α :=
+  ⟨u1.val - u2.val⟩
```

Comment:
Can these be deleted? It looks to me like you don't use them in the `Add`/`Sub` etc instances below.

Reply:
They're being used in other places, I'm going to add `HAdd` instances instead.